### PR TITLE
Farmland Trampling

### PR DESCRIPTION
**Pull request by Lucaskyy**
This PR closes: #212

### CODE REVIEW

**Comment by Lucaskyy** [comment]
Fixed the conflicts, you can merge this PR :)

**Comment by Lucaskyy** [comment]
One thing though, the farmland successfully goes away but it does NOT give your items back. This should be fixed outside of farmland since there might be other things that also have the same behavior as farmland when getting destroyed.

File: server/block/block.go [hunk 1]
```diff
@@ -216,13 +216,15 @@ func newFlammabilityInfo(encouragement, flammability int, lavaFlammable bool) Fl
 // EntityCollider is an interface for blocks with special behaviors on entity collision.
 type EntityCollider interface {
 	// EntityCollide is called on entity collision.
-	EntityCollide(e world.Entity)
+	EntityCollide(e world.Entity, pos cube.Pos)
```
Comment (T14Raptor) [inline server/block/block.go:219]: Maybe put `pos` before `e` for consistency reasons? Other block functions typically have `pos` as the first argument.

Comment (Lucaskyy) [reply server/block/block.go:219]: I agree. I'll fix that tomorrow.

Comment (Lucaskyy) [reply server/block/block.go]: Actually, I'll fix it now,

**Comment by Lucaskyy** [comment]
This PR is done and mergeable if everything is alright.
## Preview:
https://imgur.com/a/zatqrTW

File: server/player/player.go [hunk 2]
```diff
@@ -1728,7 +1733,7 @@ func (p *Player) checkBlockCollisions() {
 	w := p.World()
 
 	aabb := p.AABB().Translate(p.Position())
-	grown := aabb.Grow(0.25)
+	grown := aabb.GrowVec3(mgl64.Vec3{0, 0.25})
```
Comment (Sandertv) [inline server/player/player.go:1736]: Maybe this 0.25 needs to be changed. This is a rather big extension and I don't think we really need that. Maybe just 0.01 would even be enough.

Comment (Lucaskyy) [reply server/player/player.go:1736]: I'll try to see if that works too.

**Comment by Lucaskyy** [comment]
Note to self: use EntityLander

**Comment by T14Raptor** [comment]
I quickly finished this up and is now ready to be merged, thanks for the PR!

**Pull request merged**